scylla-cql/src/frame/frame_errors.rs

```diff
@@ -32,8 +32,10 @@ pub enum FrameDeserializationError {
     VersionNotSupported(u8),
     #[error("Connection was closed before body was read: missing {0} out of {1}")]
     ConnectionClosed(usize, usize),
-    #[error(transparent)]
-    StdIoError(#[from] std::io::Error),
+    #[error("Failed to read a frame header: {0}")]
+    HeaderIoError(Arc<std::io::Error>),
```

Comment:
nit: `a frame header` -> `the frame header` (because there's exactly one header per frame)